search-app/src/cmr/search/site/static.clj

```diff
@@ -2,33 +2,64 @@
   "The functions of this namespace are specifically responsible for returning
   ready-to-serve pages."
   (:require
+   [cheshire.core :as json]
+   [clj-http.client :as client]
    [cmr.common-app.static :as static]
-   [cmr.search.site.data :as data])
+   [cmr.common.log :refer :all]
+   [cmr.search.site.data :as data]
+   [cmr.search.site.static.directory :as directory]
+   [cmr.search.site.static.site :as site]
+   [cmr.search.site.util :as util]
+   [cmr.transmit.config :as transmit])
   (:gen-class))
 
+;; Contextual data that is used for static content in the absense of a system
+;; context, e.g., when run from the command line.
+(defrecord StaticContext [
+  ;; the deployment env; one of :prod, :uat, :sit, or :local
+  deployment-target
+  ;; the application which is being run with the context, e.g., :search
+  cmr-application
+  ;; the context of execution, e.g., :cli, when run from a system shell
+  execution-context
+  ;; a place to store context-specific data
+  static-data])
+
 (defn generate-api-docs
   "Generate CMR Search API docs."
   []
   (static/generate
    "resources/public/site/docs/search/api.html"
    "templates/search-docs-static.html"
-   (merge
-    (data/base-static)
-    {:site-title "CMR Search"
-     :page-title "API Documentation"
-     :page-content (static/md-file->html "docs/api.md")})))
+   (merge (data/base-page {:cmr-application :search
+                           :execution-context :cli})
+          {:site-title "CMR Search"
+           :page-title "API Documentation"
+           :page-content (static/md-file->html "docs/api.md")})))
 
 (defn generate-site-docs
   "Generate CMR Search docs for routes and web resources."
   []
   (static/generate
    "resources/public/site/docs/search/site.html"
    "templates/search-docs-static.html"
-   (merge
-    (data/base-static)
-    {:site-title "CMR Search"
-     :page-title "Site Routes & Web Resource Documentation"
-     :page-content (static/md-file->html "docs/site.md")})))
+   (merge (data/base-page {:cmr-application :search
+                           :execution-context :cli})
+          {:site-title "CMR Search"
+           :page-title "Site Routes & Web Resource Documentation"
+           :page-content (static/md-file->html "docs/site.md")})))
+
+(defn generate-site-resources
```

Comment:
I must have missed it. Where is this function used?

Reply:
It was used in the previous branch, when we were saving static files to the file system. I still want to keep these around, though -- very useful for performing troubleshootig checks, etc.

I'll update the docstring to that effect.

Reply:
Oh, and I'll add an entry point in `main` for this, too.